src/SFML/Network/Packet.cpp

```diff
@@ -29,12 +29,57 @@
 #include <SFML/Network/SocketImpl.hpp>
 #include <SFML/System/String.hpp>
 
+#include <algorithm>
+#include <array>
 #include <cstring>
 #include <cwchar>
 
 
 namespace sf
 {
+namespace
+{
+////////////////////////////////////////////////////////////
+/// \brief Endian test
+///
+////////////////////////////////////////////////////////////
+inline bool isBigEndian()
+{
+    int one = 1;
+    return (*reinterpret_cast<char*>(&one)) != 1;
+};
+
+
+////////////////////////////////////////////////////////////
+/// \brief Generic endianness conversion; if host is big-endian,
+/// incoming network byte order can be handled without swap, and
+/// host byte order is already network byte order
+///
+////////////////////////////////////////////////////////////
+template <typename T>
+T swapEndian(T val)
+{
+    if (isBigEndian())
+    {
+        return val;
+    }
+
+    std::array<std::uint8_t, sizeof(T)> raw;
+
+    auto* cast = reinterpret_cast<std::uint8_t*>(&val);
+    std::memcpy(raw.data(), cast, sizeof(T));
+
+    std::array<std::uint8_t, sizeof(T)> dst;
+    std::reverse_copy(raw.begin(), raw.end(), dst.begin());
+    std::memcpy(cast, dst.data(), sizeof(T));
+
+    auto* valp = reinterpret_cast<T*>(cast);
+
+    return *valp;
```

Comment:
If we're copying into the address of `val`, can't we then just return `val` whose bytes have just been rewritten? As written this appears to be starting the lifetime of a type with `reinterpret_cast` which is UB.